### ISSUE TITLE
'render_example_to_string' could not be found for ViewComponentsController

### ISSUE THREAD

**Issue by afomera** [allmarkedup/lookbook#79]
Hey again! 

**Versions**
view_component => 2.52.0
lookbook => 0.7.2
rails => 6.1.5

We recently updated to 0.7.2 in production, and our lookbook previews stopped working in production when we did so.

We're on Rails 6.1.5 right now (😭, eventually we'll be on 7.0)

I've been having trouble tracking down this error and was wondering if you had seen it, or had any helpful troubleshooting advice. I think the fix referenced https://github.com/allmarkedup/lookbook/issues/77#issuecomment-1093550028 might actually be why it stopped working for us too oddly.

<img width="1158" alt="CleanShot 2022-04-08 at 20 43 33@2x" src="https://user-images.githubusercontent.com/1741179/162553589-6a071ec1-3e88-4dfb-aa9b-78046371a600.png">

If you're curious we're running these config options in development.rb

```rb
  # development.rb
  config.view_component.preview_paths << Rails.root.join("spec", "components", "previews")
  config.view_component.show_previews_source = false
  config.view_component.preview_route = "/view_component/previews"
  config.view_component.default_preview_layout = "view_component_preview"
  config.view_component.show_previews = true
  config.lookbook.page_paths = ["docs/components"]
  config.lookbook.project_name = "Podia"
  config.lookbook.experimental_features = [:pages]
```

and in production, we had to disable the pages feature because of a Heroku deploy problem with listeners...

```rb
  # production.rb
  # ViewComponent preview configuration
  config.view_component.preview_paths << Rails.root.join("spec", "components", "previews")
  config.view_component.show_previews_source = false
  config.view_component.preview_route = "/view_component/previews"
  config.view_component.default_preview_layout = "view_component_preview"
  config.view_component.show_previews = true
  # config.lookbook.page_paths = ["docs/components"]
  config.lookbook.project_name = "Podia"
  # config.lookbook.experimental_features = [:pages]
```

We do _not_ have any controller in our app named ViewComponentsController

Here's the full stack trace if it helps:

```
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/abstract_controller/base.rb:160:in `process'
/app/vendor/bundle/ruby/2.7.0/gems/actionview-6.1.5/lib/action_view/rendering.rb:39:in `process'
/app/vendor/bundle/ruby/2.7.0/gems/lookbook-0.7.2/app/controllers/lookbook/previews_controller.rb:69:in `example_data'
/app/vendor/bundle/ruby/2.7.0/gems/lookbook-0.7.2/app/controllers/lookbook/previews_controller.rb:59:in `block in examples_data'
/app/vendor/bundle/ruby/2.7.0/gems/lookbook-0.7.2/app/controllers/lookbook/previews_controller.rb:58:in `map'
/app/vendor/bundle/ruby/2.7.0/gems/lookbook-0.7.2/app/controllers/lookbook/previews_controller.rb:58:in `examples_data'
/app/vendor/bundle/ruby/2.7.0/gems/lookbook-0.7.2/app/controllers/lookbook/previews_controller.rb:30:in `show'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/abstract_controller/base.rb:228:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/instruments/action_controller_rails_3_rails4.rb:103:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/rendering.rb:30:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/abstract_controller/callbacks.rb:42:in `block in process_action'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
/app/vendor/bundle/ruby/2.7.0/gems/react-rails-2.6.1/lib/react/rails/controller_lifecycle.rb:31:in `use_react_component_helper'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/callbacks.rb:137:in `run_callbacks'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/abstract_controller/callbacks.rb:41:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/rescue.rb:22:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/instrumentation.rb:34:in `block in process_action'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/notifications.rb:203:in `block in instrument'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/notifications.rb:203:in `instrument'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/instrumentation.rb:33:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal/params_wrapper.rb:249:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/activerecord-6.1.5/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/instruments/action_controller_rails_3_rails4.rb:120:in `process_action'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/abstract_controller/base.rb:165:in `process'
/app/vendor/bundle/ruby/2.7.0/gems/actionview-6.1.5/lib/action_view/rendering.rb:39:in `process'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal.rb:190:in `dispatch'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_controller/metal.rb:254:in `dispatch'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/route_set.rb:33:in `serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:50:in `block in serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:32:in `each'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:32:in `serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/route_set.rb:842:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/instruments/rails_router.rb:29:in `call_with_scout_instruments'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/engine.rb:539:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/railtie.rb:207:in `public_send'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/railtie.rb:207:in `method_missing'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/mapper.rb:20:in `block in <class:Constraints>'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/mapper.rb:49:in `serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:50:in `block in serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:32:in `each'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/journey/router.rb:32:in `serve'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/routing/route_set.rb:842:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/instruments/rails_router.rb:29:in `call_with_scout_instruments'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/static.rb:24:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:192:in `call!'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:169:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:192:in `call!'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:169:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:192:in `call!'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:169:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:192:in `call!'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:169:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:192:in `call!'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/strategy.rb:169:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/omniauth-1.9.1/lib/omniauth/builder.rb:45:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/metrician-0.1.0/lib/metrician/middleware/application_timing.rb:20:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/middleware.rb:17:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-attack-6.5.0/lib/rack/attack.rb:113:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/flipper-0.23.0/lib/flipper/middleware/memoizer.rb:77:in `memoized_call'
/app/vendor/bundle/ruby/2.7.0/gems/flipper-0.23.0/lib/flipper/middleware/memoizer.rb:42:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/deflater.rb:44:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/static.rb:161:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/tempfile_reaper.rb:15:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/conditional_get.rb:27:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/http/permissions_policy.rb:22:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/http/content_security_policy.rb:19:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:266:in `context'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:260:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/cookies.rb:689:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/callbacks.rb:98:in `run_callbacks'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rollbar-3.3.0/lib/rollbar/middleware/rails/rollbar.rb:25:in `block in call'
/app/vendor/bundle/ruby/2.7.0/gems/rollbar-3.3.0/lib/rollbar.rb:145:in `scoped'
/app/vendor/bundle/ruby/2.7.0/gems/rollbar-3.3.0/lib/rollbar/middleware/rails/rollbar.rb:22:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/debug_exceptions.rb:29:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rollbar-3.3.0/lib/rollbar/middleware/rails/show_exceptions.rb:22:in `call_with_rollbar'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/lograge-0.11.2/lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/rack/logger.rb:26:in `block in call'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/tagged_logging.rb:99:in `block in tagged'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/tagged_logging.rb:37:in `tagged'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/tagged_logging.rb:99:in `tagged'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/rack/logger.rb:26:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/request_store-1.5.0/lib/request_store/middleware.rb:19:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-timeout-0.6.0/lib/rack/timeout/core.rb:151:in `block in call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-timeout-0.6.0/lib/rack/timeout/support/timeout.rb:19:in `timeout'
/app/vendor/bundle/ruby/2.7.0/gems/rack-timeout-0.6.0/lib/rack/timeout/core.rb:150:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/request_id.rb:26:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rails_autoscale_agent-0.11.0/lib/rails_autoscale_agent/middleware.rb:27:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/activesupport-6.1.5/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-rewrite-1.5.1/lib/rack/rewrite.rb:24:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/executor.rb:14:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/static.rb:24:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/ssl.rb:77:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/actionpack-6.1.5/lib/action_dispatch/middleware/host_authorization.rb:142:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-cors-1.1.1/lib/rack/cors.rb:100:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/rack-2.2.3/lib/rack/content_length.rb:17:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/metrician-0.1.0/lib/metrician/middleware/request_timing.rb:35:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/scout_apm-5.1.1/lib/scout_apm/instruments/middleware_summary.rb:58:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/railties-6.1.5/lib/rails/engine.rb:539:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/configuration.rb:249:in `call'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/request.rb:77:in `block in handle_request'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/request.rb:76:in `handle_request'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/server.rb:447:in `process_client'
/app/vendor/bundle/ruby/2.7.0/gems/puma-5.5.2/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
```

Any pointers would be greatly appreciated!

**Comment by allmarkedup** [comment]
Hey @afomera - Good timing as I've actually just literally released a new beta version (0.8.0.beta.0) that makes some changes to hopefully better support deployment in production :-)

There is some documentation covering the changes here: https://github.com/allmarkedup/lookbook/tree/v0.8.0-beta.0#deployment

Could you give it a try and let me know if it solves the problems you are seeing?

**Comment by afomera** [comment]
Whoa you work fast @allmarkedup 

So I added the task enhancement as instructed, with the only difference being wrapping it in a `if Rails.env.production?` check our my end in the Rakefile.

Deployed a review instance to heroku and can confirm previews for view components do in fact work again.

However with enabling the `config.lookbook.experimental_features = [:pages]` I didn't see my docs, though I just remembered we have a `.slugignore` file... and our `/docs` folder shows up in there lol, so my bet is that's why my page documentation doesn't show up.


So TLDR: it did fix the rendering of my components in our production env, but I think something on my end prevents our page documentation from working.

**Comment by allmarkedup** [comment]
@afomera ha I'm not normally that fast, it was just fortuitous timing :-)

Great to hear that it sorted out your problem and good call on the `if Rails.env.production?` statement in the Rakefile - I'll add that into the docs I think.

I'll close this issue now but do let me know if you need any help with getting the docs working if it doesn't turn out to be just a setup issue at your end 👍

**Comment by afomera** [comment]
Thanks @allmarkedup I'm all good 👍 Moving it out of our ignored `docs/` folder make the page previews work in our production-like env.
